### ISSUE TITLE
Choose link from existing posts/pages

### ISSUE THREAD

**Issue by parfilov** [pojome/elementor#4232]
Hello! That's would be great to choose widgets' link from existing posts/pages.

**Comment by bainternet** [comment]
@parfilov 

can you elaborate? "choose widgets' link"

**Comment by redlagoon** [comment]
I think he's looking for a dropdown consisting of all posts / pages.

Posts:

```
        $this->add_control(
            'individual_post',
            [
                'label' => __( 'Select Post', 'elementor' ),
                'type' => Controls_Manager::SELECT,
                'default' => '1',
                'options' => getPostsForSelect()
            ]
        );
```

And the function:

```
    public static function getPostsForSelect( $args = null ) {
        $posts = get_posts();

        $select_posts = [];

        foreach( $posts as $post ) {
            $select_posts[$post->ID] = ucfirst( $post->post_title );
            /*
            echo $post->ID . " " . $post->post_title . " and the guid is" . $post->guid . " and the link (must build) is)" . $post->post_name;
            // Also need to check for $post->status == 'publish' and post-password == "" (empty)
            //guid is the link.
            //check if 'post_type' is post.
            echo "<br>";
            */
        }
        return $select_posts;
    }
```

As for pages, you'll need this (courtesy of @iamntz ):

```
    public static function getTermsForSelect( $tax )
    {
        $terms = get_terms([
            'taxonomy' => $tax,
            'hide_empty' => true,
        ]);

        $terms = array_map( function ( $term ) {
            return [
                $term->slug => ucfirst( $term->name ) . " ({$term->count})",
            ];
        }, $terms);

        $options = [];

        foreach( $terms as $term ) {
            $options[key( $term )] = array_values( $term )[0];
        }
        return $options;
    }
```

Where `$tax` is 'page'.

Enjoy.

**Comment by parfilov** [comment]
When I put, for example, Button widget (or other widget with link insertion opportunity), I should insert link handly. That's would be great to choose existing posts/pages as we can do in native WP editor when creating links.
![choose_link](https://user-images.githubusercontent.com/12493645/39230296-cdf38cfc-4887-11e8-9d9f-5282ea59dc73.PNG)

**Comment by parfilov** [comment]
@redlagoon that's it! But I need this natively because I have many ton-techie subscribers. Also when somthing will change in Elementor, your code may become unworkable.

**Comment by bainternet** [comment]
Thanks for the clarifications,

I merged this issue with  #255, which is on the same subject so we could deal with it more efficiently. 
Please feel free to continue the conversation there.

Thanks for your feedback!
